cmd/gateway-main.go

```diff
@@ -369,7 +369,17 @@ func gatewayMain(ctx *cli.Context, backendType gatewayBackend) {
 
 	}
 
-	globalHTTPServer = miniohttp.NewServer([]string{gatewayAddr}, registerHandlers(router, handlerFns...), globalTLSCertificate)
+	// Prepare a channel to send TLS public & private certificates
+	var keyPairCh chan miniohttp.KeyPair
+	if globalIsSSL {
+		keyPairCh = make(chan miniohttp.KeyPair)
+		go func() {
+			keyPairCh <- miniohttp.KeyPair{globalRawPublicCert, globalRawPrivateKey}
```

Comment:
Lets do these changes in a separate PR.

Reply:
> Lets do these changes in a separate PR.

Umm, I did that because it will simplify the work of the next PR since ocsp updating is little related to  hot-loading of certificates. But I can revert it since it will make reviewing easier.